### DISCUSSION TITLE
Cannot get vue-query to work with Vue 2.7

### DISCUSSION THREAD

**Discussion by santervo** [TanStack/query#6793] [Q&A]
I get error `vue-query hooks can only be used inside setup() function or functions that support injection context` when trying to use `ùseQuery` inside `<script setup>`.

I investigated it a bit, and it seems that the error comes from`useBaseQuery` checking that `getCurrentScope()` from `vue-demi` is not null or undefined. I checked that `getCurrentScope()` is defined when importing it from `vue`, but not defined when imported from `vue-demi`.

I install vue-query like this to my app:

```
import Vue from "vue";
import { VueQueryPlugin } from "@tanstack/vue-query";

Vue.use(VueQueryPlugin);
const app = new Vue({ ...}).$mount();
document.body.appendChild(app.$el);
```

In the documentation it says to use @vue/composition-api and createApp from it, but it's not used anymore with 2.7 so the documentation if probably bit outdated.

Question is: is there support for 2.7? And if there is, how should it be setup?

**Comment by DamianOsipiuk** [comment]
Could you provide reproduction? It works fine for me.

**Reply by zanlucathiago** [reply]
I'm not getting vue-query to work with Vue 2.7 during build stage... Everything worked during development with vite, but the build with rollup breaks. You can check the issue in your own sandbox, it is happening there exactly the same way it is hapening with me. Just run npm run _build (https://tanstack.com/query/v5/docs/framework/vue/examples/2.6-basic?panel=sandbox):

```
"isRef" is not exported by "node_modules/vue-demi/lib/index.mjs", imported by "node_modules/@tanstack/vue-query/build/modern/utils.js".
file: /home/projects/kkvivjkkov.github/node_modules/@tanstack/vue-query/build/modern/utils.js:2:9
1: // src/utils.ts
2: import { isRef, unref } from "vue-demi";
            ^
3: var VUE_QUERY_CLIENT = "VUE_QUERY_CLIENT";
4: function getClientKey(key) {
```

**Reply by zanlucathiago** [reply]
I posted my issue here because i tried running a post install `npx vue-demi-switch 2` and when i do that, the error is the same as @santervo pointed

**Reply by FengXiLo** [reply]
> 能否提供复现步骤？我这边运行正常。

想要在vue2.7.7的项目中使用 我是应该参考 basic 示例还是 vue2.6示例

**Reply by FengXiLo** [reply]
目前官网的沙盒环境2.6示例运行
也是报错的
